graph_weather/models/atmorep/data/dataset.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+from torch.utils.data import Dataset
+
+from graph_weather.models.atmorep.config import AtmoRepConfig
+
+
+class ERA5Dataset(Dataset):
+    def __init__(
+        self,
+        config: AtmoRepConfig,
+        data_dir: str,
+        fields=None,
+        years=None,
+        months=None,
+        transform=None,
+    ):
+        """
+        ERA5Dataset for handling atmospheric data.
+
+        Args:
+            config (AtmoRepConfig): Configuration object.
+            data_dir (str): Directory containing the data and index file.
+            fields (list, optional): List of fields to load. Defaults to config.input_fields.
+            years (list, optional): List of years to consider. Defaults to range(1979, 2022).
+            months (list, optional): List of months to consider. Defaults to 1-12.
+            transform (callable, optional): Optional transform to apply on a sample.
+        """
+        self.config = config
+        self.data_dir = data_dir
+        self.fields = fields if fields is not None else config.input_fields
+        self.transform = transform
+
+        # Expect a data index file in the data directory
+        index_file = os.path.join(data_dir, "data_index.txt")
```

Comment:
We shouldn't need a index file for dataset like this, the ARCO-ERA5 is quite simple to use and index into, so this shouldn't be necessary.